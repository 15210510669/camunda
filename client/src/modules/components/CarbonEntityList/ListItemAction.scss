@use '@carbon/styles/scss/spacing';

.CarbonListItemSingleAction > svg {
  display: block;
  margin-right: spacing.$spacing-04;
}
