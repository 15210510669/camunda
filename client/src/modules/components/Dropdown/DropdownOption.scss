.DropdownOption,
a.DropdownOption,
button.DropdownOption {
  display: block;
  white-space: nowrap;
  color: #333;
  text-decoration: none;
  padding: 6px 12px;
  font-size: 1em;
  background-color: white;
  border: none;
  display: block;
  width: 100%;
  text-align: left;
  border-radius: 0;
  
  &.is-active {
    border: none;
  }

  &:hover {
    background-color: #008fe6; /* #00a8ff -> darken 10% */
    color: #fff;
    border: none;
    transition: none;
  }

  &:focus {
    box-shadow: inset 0 0 2px 2px rgba(#00a8ff, 1);
    border: none;
  }

  &[disabled],
  &[disabled='disabled'] {
    &:hover {
      background-color: #fff;
      border: none;
    }
  }
}
