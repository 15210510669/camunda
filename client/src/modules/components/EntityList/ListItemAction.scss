.ListItemAction {
  margin: 0 14px;
  height: 32px;
  width: 32px;

  .activateButton {
    width: 32px;
    height: 32px;

    .Icon {
      width: 24px;
      height: 24px;
      font-size: 24px;
    }

    .downIcon {
      display: none;
    }
  }

  &.Dropdown.is-open .menu {
    top: 26px;
    left: auto !important; // the calculateMenuStyle method of the Dropdown sets inline styles
    right: 0;
  }

  .DropdownOption {
    padding-left: 10px;

    .Icon {
      margin-right: 8px;
    }
  }
}
