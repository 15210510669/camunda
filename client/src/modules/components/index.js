/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

export {Table} from './Table';
export {Button} from './Button';
export {Modal} from './Modal';
export {Message} from './Message';
export {Dropdown} from './Dropdown';
export {Input} from './Input';
export {Select} from './Select';
export {ButtonGroup} from './ButtonGroup';
export {BPMNDiagram, ClickBehavior} from './BPMNDiagram';
export {ControlGroup} from './ControlGroup';
export {CopyToClipboard} from './CopyToClipboard';
export {ReportRenderer} from './ReportRenderer';
export {ErrorBoundary} from './ErrorBoundary';
export {Logo} from './Logo';
export {Icon} from './Icon';
export {ActionItem} from './ActionItem';
export {ShareEntity} from './ShareEntity';
export {Switch} from './Switch';
export {Popover} from './Popover';
export {DashboardView} from './DashboardView';
export {DashboardObject} from './DashboardView';
export {DefinitionSelection} from './DefinitionSelection';
export {TargetValueBadge} from './TargetValueBadge';
export {ErrorPage} from './ErrorPage';
export {ErrorMessage} from './ErrorMessage';
export {Typeahead} from './Typeahead';
export {Labeled} from './Labeled';
export {LabeledInput} from './LabeledInput';
export {TypeaheadMultipleSelection} from './TypeaheadMultipleSelection';
export {DatePicker} from './DatePicker';
export {LoadingIndicator} from './LoadingIndicator';
export {ConfirmationModal} from './ConfirmationModal';
export {ColorPicker} from './ColorPicker';
export {EntityNameForm} from './EntityNameForm';
export {CollectionsDropdown} from './CollectionsDropdown';
export {EditCollectionModal} from './EditCollectionModal';
