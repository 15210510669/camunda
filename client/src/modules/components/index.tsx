/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

export {Button} from './Button';
export {ButtonGroup} from './ButtonGroup';
export {Tabs} from './Tabs';
export {Message} from './Message';
export {MessageBox} from './MessageBox';
export {Icon} from './Icon';
export {ActionItem} from './ActionItem';
export {Switch} from './Switch';
export {Form} from './Form';
export {Input} from './Input';
export {LabeledInput} from './LabeledInput';
export {Labeled} from './Labeled';
export {Tooltip} from './Tooltip';
export {Modal} from './Modal';
export {Dropdown} from './Dropdown';
export {Select} from './Select';
export {TextEditor} from './TextEditor';
export {PageTitle} from './PageTitle';
export {EntityDescription} from './EntityDescription';
export {EntityNameForm} from './EntityNameForm';
export {LoadingIndicator} from './LoadingIndicator';
export {DocsLink} from './DocsLink';
export {Typeahead, MultiSelect} from './Typeahead';
export {Tag} from './Tag';
export {MultiValueInput, UncontrolledMultiValueInput} from './MultiValueInput';
export {UserTypeahead} from './UserTypeahead';
export {BPMNDiagram, ClickBehavior} from './BPMNDiagram';
export {DMNDiagram} from './DMNDiagram';
export {DownloadButton} from './DownloadButton';
export {NoDataNotice} from './ReportRenderer';
export {Table} from './Table';
export {TargetValueBadge} from './TargetValueBadge';
export {DatePicker} from './DatePicker';
export {DateRangeInput} from './DateRangeInput';
export {SearchInput} from './SearchInput';
export {Checklist} from './Checklist';
export {DurationChart} from './DurationChart';
export {ColorPicker} from './ColorPicker';
export {Badge} from './Badge';
export {Popover} from './Popover';

export type {InputProps} from './Input';
export type {LabeledProps} from './Labeled';
export type {User} from './UserTypeahead';
export type {Canvas, ModdleElement, RegistryElement} from './BPMNDiagram';
export type {TooltipProps} from './Tooltip';
export type {TableBody, TableHead} from './Table';
