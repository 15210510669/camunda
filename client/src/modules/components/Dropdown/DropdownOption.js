import React from 'react';
import {Icon} from 'components';
import './DropdownOption.scss';
import classnames from 'classnames';

export default React.forwardRef(function DropdownOption({active, ...props}, ref) {
  return (
    <div
      {...props}
      onClick={evt => !props.disabled && props.onClick && props.onClick(evt)}
      className={classnames('DropdownOption', props.className, {'is-active': active})}
      tabIndex={props.disabled ? '-1' : '0'}
      ref={ref}
    >
      {props.checked && <Icon className="checkMark" type="check-small" size="10px" />}
      {props.children}
    </div>
  );
});
