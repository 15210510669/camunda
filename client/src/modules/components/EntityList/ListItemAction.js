/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';
import {Icon, Dropdown, Button, Tooltip} from 'components';

import './ListItemAction.scss';

export default function ListItemAction({actions = [], singleAction}) {
  if (!actions || actions.length === 0) {
    return <div className="ListItemAction" />;
  }

  if (singleAction) {
    const {icon, action, text} = actions[0];
    return (
      <Tooltip content={text} align="right">
        <Button
          icon
          className="ListItemAction"
          onClick={(evt) => {
            evt.preventDefault();
            action(evt);
          }}
        >
          <Icon type={icon} />
        </Button>
      </Tooltip>
    );
  }

  return (
    <Dropdown className="ListItemAction" icon label={<Icon type="context-menu" />}>
      {actions.map(({action, icon, text}, idx) => (
        <Dropdown.Option onClick={action} key={idx}>
          <Icon type={icon} /> {text}
        </Dropdown.Option>
      ))}
    </Dropdown>
  );
}
