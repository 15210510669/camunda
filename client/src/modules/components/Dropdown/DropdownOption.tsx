/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {ComponentPropsWithoutRef, forwardRef} from 'react';
import {Link} from 'react-router-dom';
import classnames from 'classnames';

import {Icon, Tooltip} from 'components';

import './DropdownOption.scss';

interface CommonProps {
  value?: string;
  active?: boolean;
  disabled?: boolean;
  checked?: boolean;
  label?: string | JSX.Element;
}

interface LinkProps extends CommonProps, Partial<ComponentPropsWithoutRef<Link>> {
  link: string;
}

interface DivProps extends CommonProps, Partial<ComponentPropsWithoutRef<'div'>> {
  link?: never;
}

export type DropdownOptionProps = LinkProps | DivProps;

export default forwardRef(function DropdownOption(
  {active, link, disabled, ...props}: DropdownOptionProps,
  ref
) {
  const commonProps = {
    ...props,
    className: classnames('DropdownOption', props.className, {'is-active': active, disabled}),
    tabIndex: disabled ? -1 : 0,
    ref,
  };

  const content = (
    <>
      {props.checked && <Icon className="checkMark" type="check-small" size="10px" />}
      {props.children}
    </>
  );

  if (link) {
    return (
      <Tooltip content={content} overflowOnly>
        <Link {...(commonProps as Partial<LinkProps>)} to={link}>
          {content}
        </Link>
      </Tooltip>
    );
  }

  return (
    <Tooltip content={content} overflowOnly>
      <div
        {...(commonProps as Partial<DivProps>)}
        onClick={(evt) => !disabled && (props as DivProps).onClick?.(evt)}
      >
        {content}
      </div>
    </Tooltip>
  );
});
