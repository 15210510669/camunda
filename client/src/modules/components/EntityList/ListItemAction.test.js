/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import {Button, Icon} from 'components';

import ListItemAction from './ListItemAction';
import DropdownOption from '../Dropdown/DropdownOption';

it('should render a placeholder when no actions are provided', () => {
  const node = shallow(<ListItemAction />);

  expect(node).toMatchSnapshot();
});

it('should render a single button if there is only a single Action in the EntityList', () => {
  const action = jest.fn();
  const node = shallow(
    <ListItemAction actions={[{icon: 'delete', text: 'Delete', action}]} singleAction />
  );

  node.find(Button).simulate('click', {preventDefault: () => {}});

  expect(action).toHaveBeenCalled();
  expect(node).toMatchSnapshot();
});

it('should render a Dropdown per default', () => {
  const node = shallow(
    <ListItemAction actions={[{icon: 'delete', text: 'Delete', action: () => {}}]} />
  );

  expect(node).toMatchSnapshot();
});

it('should render dropdown icon if passed', () => {
  let node = shallow(
    <ListItemAction actions={[{icon: 'dashboard', text: 'Delete', action: () => {}}]} />
  );

  expect(node.find(DropdownOption).find(Icon).prop('type')).toBe('dashboard');

  node = shallow(<ListItemAction actions={[{text: 'Delete', action: () => {}}]} />);

  expect(node.find(DropdownOption).find(Icon)).not.toExist();
});
