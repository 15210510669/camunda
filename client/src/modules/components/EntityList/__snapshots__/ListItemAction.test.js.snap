// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Dropdown per default 1`] = `
<Dropdown
  className="ListItemAction"
  icon={true}
  label={
    <Icon
      type="context-menu"
    />
  }
>
  <ForwardRef(DropdownOption)
    key="0"
    onClick={[Function]}
  >
    <Icon
      type="delete"
    />
     
    Delete
  </ForwardRef(DropdownOption)>
</Dropdown>
`;

exports[`should render a placeholder when no actions are provided 1`] = `
<div
  className="ListItemAction"
/>
`;

exports[`should render a single button if there is only a single Action in the EntityList 1`] = `
<Tooltip
  align="right"
  content="Delete"
>
  <ForwardRef(Button)
    className="ListItemAction"
    icon={true}
    onClick={[Function]}
  >
    <Icon
      type="delete"
    />
  </ForwardRef(Button)>
</Tooltip>
`;
